packages/babel-preset/index.js

```diff
@@ -114,5 +109,10 @@ module.exports = function shopifyCommonPreset(
       require.resolve('@babel/plugin-transform-react-constant-elements'),
   ].filter(Boolean);
 
-  return {presets, plugins};
+  const assumptions = {
```

Comment:
```suggestion
  // These assumptions are required because we use `@babel/plugin-proposal-decorators` in legacy mode
  // for compatibility with typescript
  // see https://babeljs.io/docs/en/babel-plugin-proposal-decorators#legacy
  // see https://babeljs.io/docs/en/babel-plugin-proposal-class-properties#loose
  // https://babeljs.io/docs/en/babel-plugin-proposal-private-methods#loose
  const assumptions = {
```